build-a-branch.sh

```diff
@@ -14,21 +14,30 @@
 # TODO: Make the container and log directories input arguments
 BASEDIR=$PWD
 CONTAINERDIR="/mnt/volume/test-containers"
+DEPLOYDIR="/mnt/volume/containers/"
 LOGDIR="/mnt/volume/logs"
 
 # Find out new and changed files via git diff-tree
-for FILE in `git diff-tree --no-commit-id --name-only`
+echo `git diff-tree --no-commit-id --name-only -r ${GITHUB_SHA}`
+git diff-tree --no-commit-id --name-only -r ${GITHUB_SHA}
+for FILE in `git diff-tree --no-commit-id --name-only -r ${GITHUB_SHA}`
+# Test hard-coded name
+#for FILE in ubuntu-base-image/Singularity.1804-cuda10.1
 do
-	FILENAME=basename($FILE)
+	FILENAME=`basename "$FILE"`
 	# Only operate on Singularity recipes, adopting the convention that they are all named "Singularity.appname"
-	if [[ $FILENAME =~ '^Singularity\.' ]]
+	if [[ $FILENAME =~ ^Singularity. ]]
 	then
 		# Call the resulting container and log the same as the app name
-		CONTAINERNAME=${$FILENAME#Singularity.}
-		cd dirname($FILE)
+		CONTAINERNAME=${FILENAME#Singularity.}
+		cd `dirname "$FILE"`
 		# TODO: consider the security implications of not restricting commands in sudoers or setuid the script
-		sudo singularity build $CONTAINERDIR/$CONTAINERNAME.sif $FILENAME 2>&1 > $LOGDIR/$CONTAINERNAME.log
+		sudo singularity build --force $CONTAINERDIR/$CONTAINERNAME.sif $FILENAME 2>&1 > $LOGDIR/$CONTAINERNAME.log
```

Comment:
If we build a container that already has a .sif file in place, the build will fail because singularity will prompt to overwrite it. It could occur if a recipe is pushed, builds successfully, but then a new recipe is pushed before the first one is merged and the .sif file moved.

Reply:
a) 
we need to have a line in there deleting previous artefacts like sif files. Otherwise we will fill up the disk. 

b)  we should add to CONTAINERFILENAME= the commit short sha which is $GITHUB_SHA
that way the files would be unique per build pipeline

Reply:
Do we ever want to keep a .sif file from an old commit that built but was not deployed? test-containers/ would never have anything but .sif files in it, so --force would clean up old successful test builds automatically. That said, we will need to keep $SINGULARITY_TMPDIR and $SINGULARITY_CACHEDIR clean as well.